pkg/streamer/reader.go

```diff
@@ -535,9 +474,9 @@ func (r *BinlogReader) parseFile(
 				break
 			}
 			u, _ := uuid.FromBytes(ev.SID)
-			err2 := r.advanceCurrentGtidSet(fmt.Sprintf("%s:%d", u.String(), ev.GNO))
-			if err2 != nil {
-				return errors.Trace(err2)
+			r.replaceWithHeartbeat, err = r.advanceCurrentGtidSet(fmt.Sprintf("%s:%d", u.String(), ev.GNO))
```

Comment:
could we use a local variable instead of a member of `BinlogReader`? BTW, I think we could reset `replaceWithHeartbeat` before `switch` in L446

Reply:
address in 91d1adb